### ISSUE TITLE
Direct GitLab integration v1

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#1000]
Draft announcement: https://github.com/sourcegraph/about/pull/41

Milestones:
- [x] Spec out how we are going to include JS in GitLab @ijsnow 
- [x] Write PR description that contains roadmap @ijsnow 
- [ ] mocks @francisschmaltz
- [x] [Open the PR when the description and mocks are ready](https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/23755) @ijsnow 
- [ ] Add db table to store Sourcegraph URL and access token @ijsnow 
- [ ] Implement UI for storing token and url @ijsnow
- [x] Implement CodeHost interface in GitLab @ijsnow
- [ ] Script that runs extensions using acces token and CodeHost from GitLab @ijsnow

**Comment by ijsnow** [comment]
## JS Distribution
For distribution of JS code for GitLab, we'll make a new entry point for
GitLab similar to the Phabricator extension. GitLab will pull the scripts
from the Sourcegraph instance that the User has configured Souregraph to
point at. This will allow the extension to be ran in a backwards compatible
way without all of our code having to be backwards compatible with
different versions of Sourcegraph and GitLab.

The one difference between the Phabricator extension and the forthcoming
GitLab integration's extensions is that we will take the =CodeHost=
interface implementation from GitLab rather than Sourcegraph code. This is
because it will be easier to keep the selectors and any information that
could change and break the extension closer to the code host itself. This
will also be a completely abstract implementation of handling code hosts
that will work for any new code host looking to integrate directly with
Sourcegraph.

**Comment by francisschmaltz** [comment]
@sqs Do you have a PR for the features we plan to add?

**Comment by ijsnow** [comment]
@francisschmaltz I'll be ready for (but not blocked by) mocks for storing settings soon.

I've created a table in GitLab that has an access token, sourcegraph url, and owner (user or group/org). The token is required for the initial implementation due to CORS on Sourcegraph. The owner can be optional (global for a GitLab instance) and the URL will just be sourcegraph.com to start with.

**Comment by nicksnyder** [comment]
Backlogging this

**Comment by felixfbecker** [comment]
GitLab merge request: https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/30574

**Comment by felixfbecker** [comment]
Closing in favor of #5380
